src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3125,8 +3125,23 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                                       ASR::down_cast<ASR::List_t>(type)->m_type, nullptr);
                 return false;
             } else if (ASR::is_a<ASR::Tuple_t>(*type)) {
+                int i = 0;
                 index = ASRUtils::EXPR(tmp);
-                int i = ASR::down_cast<ASR::IntegerConstant_t>(ASRUtils::EXPR(tmp))->m_n;
+                ASR::expr_t* val = ASRUtils::expr_value(index);
+                if (val && ASR::is_a<ASR::IntegerConstant_t>(*val)) {
+                    i = ASR::down_cast<ASR::IntegerConstant_t>(val)->m_n;
+                    int tuple_size =  ASR::down_cast<ASR::Tuple_t>(type)->n_type;
+                    if (i < 0) {
+                        i = tuple_size + i;
+                        ASR::ttype_t *int_type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
+                                                        4, nullptr, 0));
+                        index = ASRUtils::EXPR(ASR::make_IntegerConstant_t(
+                            al, loc, i, int_type));
+                    }
+                    if (i >= tuple_size || i < 0) {
+                        throw SemanticError("TupleIndex out of range", loc);
```

Comment:
Could you make this to `"Tuple index out of range"`.
I just observed that the `IndexError` messages in `Lpython` are not uniform depending on who added the code.
For `Strings` I believe it is tackled like this -
```
if(idx1 > s_len || idx1 <= (-1*s_len)){
        printf("String index out of Bounds\n");   # A simple print statement
        exit(1);
    }
```
For list it is handled like this -
```
std::string index_error = "IndexError: %s%d%s%d\n",
            message1 = "List index is out of range. Index range is (0, ",
            message2 = "), but the given index is ";

(lp) C:\Users\kunni\lpython>src\bin\lpython try1.py
IndexError: List index is out of range. Index range is (0, 4), but the given index is -5    # Example
```
So here `IndexError` is printed and a bunch of unnecessary information is given. And in both cases I don't see any `SemanticError` being thrown.

Reply:
In python the error is quite simple and straightforward for everything- 
`IndexError: 'data_type' index out of range`

@certik What do you feel about this . If we set a standard here for `tuple` then I can make changes in `str` and `list`

Reply:
I would say using something like `'data_type' index out of bounds` would make sense .... as in we could standardize the messages . 
1) I would say we could go for `bounds` rather than `range` ( range represents something from 0 to n-1 I guess)
2) We don't neccesaarily have to specify the given index which is falling outside bounds , like we do for lists like ..... `but the given index is -5`

Reply:
cc @certik